recipe-server/client/control_new/components/recipes/CloneRecipePage.js

```diff
@@ -0,0 +1,117 @@
+import { message } from 'antd';
+import autobind from 'autobind-decorator';
+import { Map } from 'immutable';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link, push as pushAction } from 'redux-little-router';
+
+import { SimpleLoadingOverlay } from 'control_new/components/common/LoadingOverlay';
+import RecipeForm from 'control_new/components/recipes/RecipeForm';
+import QueryRecipe from 'control_new/components/data/QueryRecipe';
+
+import { createRecipe as createAction } from 'control_new/state/recipes/actions';
+import { getRecipeFromURL } from 'control_new/state/recipes/selectors';
+
+@connect(
+  state => {
+    const {
+      isLatestRevision,
+      recipe,
+      recipeId,
+      revisionId,
+    } = getRecipeFromURL(state);
```

Comment:
So I went back and looked at what I was doing on the recipe details page and on further thought I think this logic is actually clean enough that it doesn't need another selector or a "container":
```javascript
    const recipeId = getUrlParamAsInt(state, 'recipeId');
    const latestRevisionId = getLatestRevisionIdForRecipe(state, recipeId, '');
    const revisionId = getUrlParam(state, 'revisionId', latestRevisionId);
    const revision = getRevision(state, revisionId, new Map());
    const isLatestRevision = isLatestRevision(state, revisionId);
```